testng-core/src/main/java/org/testng/internal/invokers/MethodInvocationHelper.java

```diff
@@ -417,6 +423,18 @@ private static void invokeWithTimeoutWithNewExecutor(
     }
   }
 
+  private static StackTraceElement[] getRunningMethodStackTrace(ExecutorService exec) {
+    if (!(exec instanceof ThreadPoolExecutor)) return null;
+    ThreadFactory threadFactory = ((ThreadPoolExecutor) exec).getThreadFactory();
+
+    if (!(threadFactory instanceof TestNGThreadFactory)) return null;
+    TestNGThreadFactory testNGThreadFactory = (TestNGThreadFactory) threadFactory;
+
+    Collection<Thread> threads = testNGThreadFactory.getRunningThreads();
+    if (threads.size() == 1) return threads.iterator().next().getStackTrace();
```

Comment:
Can we please wrap all these simple conditions within a `{}` for the sake of readability ?